rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -286,6 +286,10 @@ private int tryDoEnd(boolean compensate) {
             int twoPhaseOutcome = retryGetEndStatus(endPath, compensate);
 
             if (twoPhaseOutcome != -1) {
+                if (parentId != null && status == ParticipantStatus.Completed) {
+                    return forget(null);
```

Comment:
also if there is no status method in the participant the retryGetEndStatus returns -1 so this seems that it will never get executed.

Reply:
But there might be a status method and it might have been called and returned Completed.